### PR TITLE
feat(scripts): pull pdfs from legal-docs

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- We need to pull the latest pdfs from legal-docs into the fxa assets folder.

## This pull request

- Create a script to copy the latest pdfs into assets/legal.
- Create a Github Action to automatically pull in the pdfs from legal-docs, and open a pull request to move pdfs into assets/legal.

## Issue that this pull request solves

Closes: #fxa-5100

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: _scripts/pull-legal-docs.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  const myArgs = process.argv.slice(2);
+
+  sourceDirectory = path.normalize(myArgs[0] || DEFAULT_SOURCE_DIRECTORY);
+  targetDirectory = path.normalize(myArgs[1] || DEFAULT_TARGET_DIRECTORY);
+
+  try {
+    await checkDir(targetDirectory);
+  } catch (err) {
+    throw new Error(
+      `The provided target directory does not exists. (${targetDirectory})`
```
Comment (sardesam) [inline _scripts/pull-legal-docs.js:61]: Nit: change 'exists' to 'exist'.

File: _scripts/pull-legal-docs.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    );
+  }
+  try {
+    await checkDir(sourceDirectory);
+  } catch (err) {
+    throw new Error(
+      `The provided source directory does not exists. (${sourceDirectory})`
```
Comment (sardesam) [inline _scripts/pull-legal-docs.js:68]: Nit: change 'exists' to 'exist'.

**Review by sardesam — APPROVED** [review]
r+wc

LGTM! 👍 Two grammar nits in comments but that's it! Thank you for taking care of this. 🙂

Edit: removed a question that was no longer relevant.

File: .github/workflows/pull-legal-docs.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          path: fxa
+          fetch-depth: 2
+
+      - name: Clone legal-docs repository
+        uses: actions/checkout@v3
+        with:
+          repository: mozilla/legal-docs
+          ref: prod
+          path: legal-docs
+          token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (flodolo) [inline .github/workflows/pull-legal-docs.yml:23]: I don't think you need the secret to pull from legal-docs (public repo)?

Comment (StaberindeZA) [reply .github/workflows/pull-legal-docs.yml:23]: That's what I was expecting as well, but I kept running into errors similar to the one in [this failed run](https://github.com/mozilla/fxa/actions/runs/3274794590/jobs/5388794337).

Comment (flodolo) [reply .github/workflows/pull-legal-docs.yml:23]: Weird. Not sure if the workflow link is correct in that page, but the workflow has a `token` attribute. I assume the error means "you provide a token attribute but it's not in the repo's settings" more than "you need to provide a token".

With that said, if it works, I probably wouldn't spend more time on it. Debugging actions is not fun.

**Pull request merged**